web/src/app/dialogs/components/DropDownMenu.js

```diff
@@ -1,65 +1,61 @@
-import React, { Component } from 'react'
+import React, { useState } from 'react'
+import { PropTypes as p } from 'prop-types'
 import IconButton from '@material-ui/core/IconButton'
 import Menu from '@material-ui/core/Menu'
 import MenuItem from '@material-ui/core/MenuItem'
-import withStyles from '@material-ui/core/styles/withStyles'
 import { MoreVert as MoreVertIcon } from '@material-ui/icons'
-import { styles } from '../../styles/materialStyles'
 
 /*
   Takes a list of options each with a label and an onClick function
 */
 
-@withStyles(styles)
-export default class DropdownMenu extends Component {
-  constructor(props) {
-    super(props)
+function DropDownMenu(props) {
+  const [anchorEl, setAnchorEl] = useState(null)
+  const { options, onExited } = props
 
-    this.state = {
-      anchorEl: null,
-    }
-  }
-
-  render() {
-    const options = this.props.options
+  return (
+    <div>
+      <IconButton
+        aria-label={props['aria-label']}
+        data-cy={props['data-cy']}
+        onClick={(event) => setAnchorEl(event.currentTarget)}
+        aria-haspopup='true'
+        style={{ color: props.color || 'inherit' }}
+      >
+        <MoreVertIcon />
+      </IconButton>
+      <Menu
+        anchorEl={anchorEl}
+        open={!!anchorEl}
+        onExited={onExited}
+        onClose={() => {
+          setAnchorEl(null)
+        }}
+      >
+        {options.map((option) => (
+          <MenuItem
+            key={option.label}
+            onClick={() => {
+              setAnchorEl(null)
+              option.onClick()
```

Comment:
This was intended to be fired upon `onExited`, after animations have finished. IIRC we had a lot of test flakiness leading to it being implemented this way.

Might require a little creativity to get it to work with hooks. Or some careful thought/validation to see if it's still required.

Reply:
I'd be curious to see how this runs against our tests as is, without the use of classes/old setState

Reply:
IIRC it was a focus-stealing issue -- dialog opening before the menu closed or something like that

Reply:
vague memories of some nightmarish test debugging come to mind

Reply:
This component isn't being utilized anywhere but I ran through some cypress scenarios locally and everything seemed to work as expected with the latest changes.